src/Conf.ml

```diff
@@ -60,170 +78,356 @@ end
+    let doc =
+      "Attempt to generate output which does not change (much) when \
+       post-processing with ocp-indent. Can be set in a config file with a \
+       `ocp-indent-compat = {true,false}` line."
+    in
+    let env = Arg.env_var "OCAMLFORMAT_OCP_INDENT_COMPAT" in
+    let names = ["ocp-indent-compat"] in
+    let default = false in
+    C.flag ~names ~default ~env ~doc ~allow_inline:true ~update:
+      (fun conf x -> {conf with ocp_indent_compat= x} )
+
+  let parens_tuple =
+    let doc =
+      "Parens tuples. Can be set in a config file with a `parens-tuple = \
+       {multi-line-only,always}` line. `multi-line-only` mode will try to \
+       skip parens for single-line tuples."
+    in
+    let env = Arg.env_var "OCAMLFORMAT_PARENS_TUPLE" in
+    let name = "parens-tuple" in
+    let all =
+      [("multi-line-only", `Multi_line_only); ("always", `Always)]
+    in
+    let default = `Always in
+    C.choice ~name ~all ~default ~env ~doc ~allow_inline:true ~update:
+      (fun conf x -> {conf with parens_tuple= x} )
+
+  let sparse =
+    let doc =
+      "Generate more sparsely formatted code. Can be set in a config file \
+       with a `sparse = {true,false}` line."
+    in
+    let env = Arg.env_var "OCAMLFORMAT_SPARSE" in
+    let default = false in
+    C.flag ~names:["sparse"] ~default ~doc ~env ~allow_inline:true ~update:
+      (fun conf x -> {conf with sparse= x} )
+
+  let wrap_comments =
+    let doc =
+      "Wrap comments and docstrings. Comments and docstrings are divided \
+       into paragraphs by open lines (two or more consecutive newlines), \
+       and each paragraph is wrapped at the margin. Multi-line comments \
+       with vertically-aligned asterisks on the left margin are not \
+       wrapped. Consecutive comments with both left and right margin \
+       aligned are not wrapped either. Can be set in a config file with a \
+       `wrap-comments = {false,true}` line."
+    in
+    let env = Arg.env_var "OCAMLFORMAT_WRAP_COMMENTS" in
+    let default = false in
+    C.flag ~names:["wrap-comments"] ~default ~doc ~env ~allow_inline:true
+      ~update:(fun conf x -> {conf with wrap_comments= x} )
+
+  let type_decl =
+    let doc =
+      "Style of type declaration. Can be set in a config file with a \
+       `type-decl = {compact,sparse}` line. `compact` will try to format \
+       constructors and records definition in a single line. `sparse` will \
+       always break between constructors and record fields."
+    in
+    let env = Arg.env_var "OCAMLFORMAT_TYPE_DECL" in
+    let name = "type-decl" in
+    let all = [("compact", `Compact); ("sparse", `Sparse)] in
+    let default = `Compact in
+    C.choice ~name ~all ~default ~env ~doc ~allow_inline:true ~update:
+      (fun conf x -> {conf with type_decl= x} )
+
+  let margin =
+    let docv = "COLS" in
+    let doc =
+      "Format code to fit within $(docv) columns. Can be set in a config \
+       file with a `margin = COLS` line."
```

Comment:
This should mention that it it not supported in attributes.

Reply:
done